pkg/client/vcwallet/client.go

```diff
@@ -9,25 +9,139 @@ package vcwallet
+		opts.passphrase = passphrase
+	}
+}
+
+// WithKeyServerURL option, when provided then wallet client will use remote kms for key operations.
+// This option will be ignore if provided with 'WithSecretLockService' option.
+func WithKeyServerURL(url string) KeyManagerOptions {
+	return func(opts *kmsOpts) {
+		opts.keyServerURL = url
+	}
 }
 
 // Client enable access to verifiable credential wallet features.
 type Client struct {
 	// ID of wallet content owner
 	userID string
+
+	// wallet profile
+	profile *profile
+
+	// storage provider
+	storeProvider storage.Provider
+}
+
+// Get returns new verifiable credential wallet client for given user.
+// returns error if wallet profile is not found.
+// To create a new client with new wallet profile, use `New()`.
+func Get(userID string, ctx provider) (*Client, error) {
+	store, err := newProfileStore(ctx.StorageProvider())
+	if err != nil {
+		return nil, fmt.Errorf("failed to get store to fetch VC wallet profile info: %w", err)
+	}
+
+	profile, err := store.get(userID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get VC wallet profile: %w", err)
+	}
+
+	return &Client{userID: userID, profile: profile, storeProvider: ctx.StorageProvider()}, nil
+}
+
+// Create creates and returns new verifiable credential wallet client by creating new wallet profile for given user.
+// returns error if wallet profile is already created.
+// Use `Get()` for already created VC wallet profile.
+// Use `Update()` for replacing an already created VC wallet profile.
+func Create(userID string, ctx provider, options ...KeyManagerOptions) (*Client, error) {
+	return createOrUpdate(userID, ctx, false, options...)
+}
+
+// Update creates and returns new verifiable credential wallet client by creating or replacing
+// wallet profile for given user.
+// Use `Get()` for already created VC wallet profile.
+// Caution: you might lose your existing keys if you change kms options.
+func Update(userID string, ctx provider, options ...KeyManagerOptions) (*Client, error) {
+	return createOrUpdate(userID, ctx, true, options...)
+}
+
+func createOrUpdate(userID string, ctx provider, update bool, options ...KeyManagerOptions) (*Client, error) {
+	opts := &kmsOpts{}
+
+	for _, opt := range options {
+		opt(opts)
+	}
+
+	profile, err := createProfile(userID, opts.passphrase, opts.secretLockSvc, opts.keyServerURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create new VC wallet client: %w", err)
+	}
+
+	store, err := newProfileStore(ctx.StorageProvider())
+	if err != nil {
+		return nil, fmt.Errorf("failed to get store to save VC wallet profile: %w", err)
+	}
+
+	err = store.save(profile, update)
+	if err != nil {
+		return nil, fmt.Errorf("failed to save VC wallet profile: %w", err)
+	}
+
+	return &Client{userID: userID, profile: profile, storeProvider: ctx.StorageProvider()}, nil
+}
+
+// Open unlocks wallet client's key manager instance and returns a token for subsequent use of wallet features.
```

Comment:
Open vs Get is confusing.

Reply:
package level functions:
`vcwallet.Get()` to get already created wallet.
`vcwallet.Create()` to get wallet client by creating new profile.
`vcwallet.Update()` to get wallet client by updating existing profile.

Open() to be called on wallet client instance.
`walletInstance.Open()`

Reply:
fixed as part of https://github.com/hyperledger/aries-framework-go/pull/2552#discussion_r579813776